build.gradle

```diff
@@ -1,380 +1,439 @@
+             events = mutableSetOf(
+            //     TestLogEvent.PASSED,
+            //     TestLogEvent.SKIPPED,
+                 TestLogEvent.FAILED,
+            //     TestLogEvent.STANDARD_OUT,
+            //     TestLogEvent.STANDARD_ERROR
+             )
+
             // Show the found unexpected diagnostics and expected diagnostics not found.
-            exceptionFormat "full"
+            exceptionFormat = TestExceptionFormat.FULL
         }
 
         // After each test, print a summary.
-        afterSuite { desc, result ->
-            if (desc.getClassName() != null) {
-                long mils = result.getEndTime() - result.getStartTime()
-                double seconds = mils / 1000.0
-
-                println "Testsuite: ${desc.getClassName()}\n" +
+        // See https://github.com/gradle/kotlin-dsl/issues/836
+        addTestListener(object : TestListener {
+            override fun beforeSuite(suite: TestDescriptor) {}
+            override fun beforeTest(testDescriptor: TestDescriptor) {}
+            override fun afterTest(testDescriptor: TestDescriptor, result: TestResult) {}
+
+            override fun afterSuite(desc: TestDescriptor, result: TestResult) {
+                if (desc.className != null) {
+                    val mils = result.endTime - result.startTime
+                    val seconds = mils / 1000.0
+
+                    logger.info(
+                        "Testsuite: ${desc.className}\n" +
                         "Tests run: ${result.testCount}, " +
                         "Failures: ${result.failedTestCount}, " +
                         "Skipped: ${result.skippedTestCount}, " +
-                        "Time elapsed: ${seconds} sec\n"
+                        "Time elapsed: $seconds sec\n"
+                    )
+                }
             }
-        }
+        })
     }
 }
 
+// TODO: convert the commented out code to Kotlin
 /// Commented out because plugins section is commented out
 // /* Configuration for formatting */
 // googleJavaFormat {
-//   // toolVersion '1.3'
-//   options style: 'AOSP'
+//   // toolVersion "1.3"
+//   options style: "AOSP"
 // }
 // tasks.googleJavaFormat {
-//   group 'Formatting'
+//   group "Formatting"
 //   description = "Reformat Java source code with Google-Java-format"
-//   exclude 'testing'
-//   exclude 'testinputs'
+//   exclude "testing"
+//   exclude "testinputs"
 // }
 // tasks.verifyGoogleJavaFormat {
-//   group 'Formatting'
+//   group "Formatting"
 //   description = "Check Java source code is in Google-Java-format"
-//   exclude 'testing'
-//   exclude 'testinputs'
+//   exclude "testing"
+//   exclude "testinputs"
 // }
 
-task etags {
+val etags by tasks.registering {
     doLast {
-        def sources = (sourceSets.main.java).getFiles().collect({ src -> src.getPath() }).sort()
-        def sourcesStr = sources.inject(null, { acc, source -> acc ? acc + " " + source : source })
-
-        def proc = "etags ${sourcesStr} ".execute()
```

Comment:
`java.io.IOException: Cannot run program "etags": error=2, No such file or directory` seems to be an existing issue.